Presentation.Web/app/services/exportGridToExcelService.ts

```diff
@@ -77,6 +103,55 @@
             }
             return template;
         }
+
+        private checkIfAllColumnsAreHidden(columns: IKendoGridColumn<any>[]): boolean {
+            let isHidden = true;
+            columns.forEach(column => {
+                if (!column.hidden) {
+                    isHidden = false;
+                    return;
+                }
+            });
+
+            return isHidden;
+        }
+
+        private showSelectedColumns(columns: IKendoGridColumn<any>[], e: IKendoGridExcelExportEvent<any>) {
+            _.forEach(this.columnsToShow,
+                column => {
+                    if (column.parentId === undefined)
+                        return;
+
+                    const columnToShow = columns.filter(x => x.persistId === column.columnId)[0];
+                    if (columnToShow === undefined || columnToShow === null)
+                        return;
+
+                    columnToShow.tempVisual = true;
+                    e.sender.showColumn(columnToShow);
+                }
+            );
+        }
+
+        private sortColumnArray() {
+            this.columnsToShow.forEach((column, i)=> {
+                if (column.parentId === undefined)
+                    return;
+
+                var parentColumn = this.columnsToShow.filter(x => x.columnId === column.parentId)[0];
```

Comment:
what if parent could not be found?
Then it will break rendering :-)
Perhaps log an error to console and then throw an error